### DISCUSSION TITLE
Sbt compile cannot include the dependency jar to classpath

### DISCUSSION THREAD

**Discussion by ethan2000** [sbt/sbt#7530] [Q&A]
steps
>
1.Add new resolver such as 

```scala
resolvers ++= Seq(
  "github-package" at "https://maven.pkg.github.com/xxx/*",
)
```
2.Add the dependency like this
```scala
libraryDependencies ++= Seq(
    "com.xxx.yyy" %% "groupid" % "0.21.1",
)
```

problem
>
When I run `sbt compile` the dependency jar from that repo can be download successfully, but finally it will be failed and it will pop this error: object xxx is not memeber of the package...... I am sure the class(com.xxx.AuditEvent) is under the dependency jar.
Error Info:
```
[info] compiling 67 Scala sources to /Users/ethan/global/orgs/target/scala-2.12/classes ...
[error] /Users/ethan/com.test.AuditService.scala:3:23: object audit is not a member of package com.xxx
[error] import com.xxx.AuditEvent
```
note
>
sbt is 1.9.8 and scala 2.12

Workaround
>
If I copy the dependency jar from 
Users/ethan/Library/Caches/Coursier/v1/https/maven.pkg.github.com/com/xxx/1.0.6/xxx.jar
to
Users/ethan/Library/Caches/Coursier/v1/https/repo1.maven.org/com/xxx/1.0.6/xxx.jar
It can work well, so I guess that , sbt can add the jars from https/repo1.maven.org(default resolver) to classpath but can not add the new resolver repo such as `maven.pkg.github.com(private resolver)` to the classpath.

**Comment by SethTisue** [comment]
Is there a sample dependency we could try this with?

**Comment by ethan2000** [comment]
> Is there a sample dependency we could try this with?

Because the maven.pkg.github.com is private resolver and it is github package, so the dependency jar is  from another project under our company. 

I am aware of that all the dependency jar will cached to the same directory by old cache strategy(ivy2), but for couriser it will split to different directory, such as (maven.pkg.github.com / repo1.maven.org / ...). So I found the dependency classes cannot be found under the directory named maven.pkg.github.com.

And as I said ,I am confused that copy the dependency jar to the repo1.maven.org directory, it can work.

**Comment by SethTisue** [comment]
Is `dependencyClasspath` as expected?

**Comment by ethan2000** [comment]
> dependencyClasspath

Yes

**Comment by ethan2000** [comment]
That is a sample:
libraryDependencies ++= Seq(
  "com.companyname"              % "business-client-java11"          % "1.55.6", 
  "org.springframework.retry"    % "spring-retry"     % "1.3.4",
  "io.dropwizard.metrics"        % "metrics-core"     % "4.2.19",
)

"com.companyname"              % "business-client-java11"          % "1.55.6", 
This dependency will be cached to ~/Library/Caches/Coursier/v1/https/maven.pkg.github.com

other dependencies will be cached to ~/Library/Caches/Coursier/v1/https/repo1.maven.org


You know the order of downloading dependencies is repo1.maven.org, and then from https/maven.pkg.github.com.
So some of them can be downloaded from public repo,but business-client-java11 only can be download from https/maven.pkg.github.com, because this our private repo.

I checked the dependencyClasspath, business-client-java11 is included.

**Comment by SethTisue** [comment]
You might run sbt with `-debug` to see what actual compiler invocation `compile` is doing.

**Comment by SethTisue** [comment]
I'm going to convert this to a discussion, as we don't yet have a way of reproducing this on anyone's computer but yours, and we don't have yet clear evidence that the root cause is a bug in sbt.

**Reply by ethan2000** [reply]
@SethTisue I compile with debug, and no more error info about this.

[debug] 	-classpath
I checked this, it also include the jar I needed.
But same error :
object audit is not a member of package com.trade.audit
